### PR TITLE
[Data Logger] Include `build_logger` in the server logic

### PR DESCRIPTION
**Pull request by dbogunowicz**
Changes tested manually

### CODE REVIEW

File: src/deepsparse/server/helpers.py [hunk 1]
```diff
@@ -11,86 +11,28 @@
...
-]
 
+__all__ = ["log_system_info", "default_logger"]
 
 _LOGGER = logging.getLogger(__name__)
-_SUPPORTED_LOGGERS = {"prometheus": PrometheusLogger}
 
 
-def default_logger_manager() -> ManagerLogger:
+def default_logger() -> PythonLogger:
```
Comment (corey-nm) [inline src/deepsparse/server/helpers.py:28]:
```suggestion
def default_logger() -> BaseLogger:
```

File: src/deepsparse/server/helpers.py [hunk 2]
```diff
@@ -11,86 +11,28 @@
...
-            raise ValueError(
-                f"Unknown logger integration {integration}. Supported integrations: "
-                f"{list(_SUPPORTED_LOGGERS)}"
-            )
-        logger = logger_class(**logger_kwargs)
-        _LOGGER.info(f"Created logger {logger}")
-        loggers.append(logger)
 
-    return ManagerLogger(loggers=loggers)
+def log_system_info(server_logger: BaseLogger):
```
Comment (corey-nm) [inline src/deepsparse/server/helpers.py:37]: 👍

File: src/deepsparse/server/server.py [hunk 3]
```diff
@@ -208,42 +206,32 @@ def _add_endpoint(
...
     context: Context,
-    pipeline_logger: ManagerLogger,
+    server_logger: BaseLogger,
 ):
     pipeline_config = endpoint_config.to_pipeline_config()
     pipeline_config.kwargs["executor"] = executor
 
     _LOGGER.info(f"Initializing pipeline for '{endpoint_config.name}'")
-    pipeline = Pipeline.from_config(pipeline_config, context)
+    pipeline = Pipeline.from_config(pipeline_config, context, server_logger)
```
Comment (corey-nm) [inline src/deepsparse/server/server.py:215]: i think this is fine for now, but still think we should have separate loggers for each pipeline. can do in a separate PR

File: src/deepsparse/server/server.py [hunk 4]
```diff
@@ -208,42 +206,32 @@ def _add_endpoint(
...
-            pipeline_outputs,
-            pipeline_inputs,
-            engine_inputs,
-            inference_timing,
-        ) = pipeline.run_with_monitoring(request)
-        pipeline_logger.log_latency(pipeline_name, inference_timing)
-        pipeline_logger.log_data(
-            pipeline_name, (pipeline_inputs, engine_inputs), pipeline_outputs
-        )
+        pipeline_outputs = pipeline(request)
```
Comment (corey-nm) [inline src/deepsparse/server/server.py:231]: so much cleaner 🔥

File: src/deepsparse/server/server.py [hunk 5]
```diff
@@ -208,42 +206,32 @@ def _add_endpoint(
...
+        server_logger = pipeline.logger
+        if server_logger:
+            log_system_info(server_logger)
```
Comment (corey-nm) [inline src/deepsparse/server/server.py:234]: fine for now, i think we may want to move this into a background thread in the future. e.g. if server doesn't receive a request for an hour we still probablky want to be logging system info during that time

**Review by corey-nm — APPROVED** [review]
lgtm, can adjust whether we have single logger or multiple per pipeline in another PR

File: src/deepsparse/server/config.py [hunk 6]
```diff
@@ -166,6 +166,28 @@ class ServerConfig(BaseModel):
         ),
     )
 
+    @validator("endpoints")
```
Comment (rahul-tuli) [inline src/deepsparse/server/config.py:169]: nice!

File: src/deepsparse/server/build_logger.py [hunk 7]
```diff
@@ -93,9 +97,7 @@ def build_function_loggers(
             continue
         for target_logging_cfg in endpoint.data_logging:
             target = target_logging_cfg.target
-            target_identifier = _get_target_identifier(
-                endpoint.name or endpoint.task, target
-            )
+            target_identifier = _get_target_identifier(endpoint.name, target)
```
Comment (rahul-tuli) [inline src/deepsparse/server/build_logger.py:100]: I'm assuming this is safe now due to the validator added to `EndPointConfig`?

**Review by bfineran — CHANGES_REQUESTED** [review]
LGTM - let's use this PR to update our E2E tests

File: src/deepsparse/server/server.py [hunk 8]
```diff
@@ -140,7 +143,7 @@ def _health():
     def _add_endpoint_endpoint(cfg: EndpointConfig):
         if cfg.name is None:
             cfg.name = f"endpoint-{len(app.routes)}"
-        _add_endpoint(app, server_config, cfg, executor, context, pipeline_logger)
+        _add_endpoint(app, server_config, cfg, executor, context, server_logger)
```
Comment (rahul-tuli) [inline src/deepsparse/server/server.py:146]: nit: could we now change these to keyword arguments? like the following: 
```
_add_endpoint(app=app, server_config=server_config, ...)
```

My train of thought being that with this, the order of calling arguments would not matter and reduce potential bugs?

I also feel it provides a first level of documentation on how to call a function.

But this is a personal preference and is not a blocker

File: src/deepsparse/server/server.py (hunk 5 above)
Comment (rahul-tuli) [reply src/deepsparse/server/server.py:234]: +1 that is a great callout

File: src/deepsparse/server/server.py (hunk 4 above)
Comment (rahul-tuli) [reply src/deepsparse/server/server.py:231]: +1

File: src/deepsparse/pipeline.py [hunk 9]
```diff
@@ -541,6 +542,8 @@ def from_config(
             other runtime information that will be used across instances of the
             MultiModelEngine to provide optimal performance when running
             multiple models concurrently
+        :param logger: An optional DeepSparse Logger object for inference
```
Comment (rahul-tuli) [inline src/deepsparse/pipeline.py:545]: nit: DeepSparse Base Logger?

File: src/deepsparse/server/build_logger.py (hunk 7 above)
Comment (dbogunowicz) [reply src/deepsparse/server/build_logger.py:100]: yessir

File: src/deepsparse/pipeline.py (hunk 9 above)
Comment (dbogunowicz) [reply src/deepsparse/pipeline.py:545]: I mean, some kind of logger. Every DeepSparse Logger will inherit from the Base Logger.

File: src/deepsparse/server/config.py [hunk 10]
```diff
@@ -191,14 +191,36 @@ class ServerConfig(BaseModel):
 
     endpoints: List[EndpointConfig] = Field(description="The models to serve.")
 
-    loggers: Dict[str, Optional[Dict[str, Any]]] = Field(
+    loggers: Dict[str, Dict[str, Any]] = Field(
```
Comment (dbogunowicz) [inline src/deepsparse/server/config.py:194]:
```suggestion
    loggers: Dict[str, Optional[Dict[str, Any]]] = Field(
```

**Comment by dbogunowicz** [comment]
All the server tests are locally green except for:
- `test_prometheus` - obviously, since Prometheus "does not exist yet" from this branches POV
- `test_hot_reloading.py::test_task_cli_hot_reload` - we have established with Corey that this may be something that might be worth taking a look once we have GHA running on the branch (so once this lands in the feature branch).

**Review by bfineran — APPROVED** [review]
